src/screens/Forum/ForumHomeScreen.js

```diff
@@ -15,46 +16,74 @@ export default class ForumHomeScreen extends React.Component {
       posts: [],
       loading: true,
     };
+    this.navigateToPostScreen = this.navigateToPostScreen.bind(this);
   }
 
   componentDidMount() {
-    firestore().collection('forum_posts')
-      .get()
-      .then((snapshot) => {
+    this.unsubscribe = auth().onAuthStateChanged((user) => {
```

Comment:
You can rename this from `this.unsubscribe` to `this.unsubscribeFromAuth` and `this.snapshotUnsubscribe` to `this.unsubscribeFromFirestore` so we know what each unsubscribe function does.